src/node/stage/approved.rs

```diff
@@ -624,7 +653,9 @@ impl Approved {
         // With the removal of parsec, sending parsec_gossip won't update the peer anymore.
         // We will have to send the message later on to allow the peer be updated by other delayed
         // messages.
-        // TODO: use other duration schedule.
+        // TODO: When the testing framework is updated to use real messaging underlayer, i.e. not
+        // using fake clock, the bounced unknown message could be resent immediately. As the network
+        // already took time to exchange messages, which allows the lagging node to be updated.
         self.gossip_timer_token = Some(core.timer.schedule(self.consensus_engine.gossip_period()));
```

Comment:
I would also remove this gossip line and we will test this with real network if necessary. We cannot have test code in our production code. So I would go for the real deal here as well.

Reply:
I agree with not having test code in the production code. 
However I think that will be out of the scope of this PR.

Another option is put this under a feature flag, but I think it shall be avoided as well?

Maybe the PR of using real deal for test shall be priorized ?

Reply:
Yes I agree, we must just get well away from mock as it's cluttering up logic now. It's actually worse here as it's forcing us to adopt logic we defo do not want. Fake time is a real problem for logic in decentralised network so we must get away from it very fast. 
I think we need to get parsec gone, go async and then do real tests as we cannot rely on the test suite for real world validation. At least we are all on the same boat here and if mock is used at all it will be restricted to unit tests.